internal/store/access_role.go

```diff
@@ -32,7 +32,11 @@ type CloudProviderAccessRoleLister interface {
 	CloudProviderAccessRoles(string) (*atlas.CloudProviderAccessRoles, error)
 }
 
-// CreateCloudProviderAccessRole encapsulate the logic to manage different cloud providers
+type CloudProviderAccessRoleDisabler interface {
+	DisableCloudProviderAccessRoles(*atlas.CloudProviderDeauthorizationRequest) error
+}
```

Comment:
[nit] I'd like to keep this in line with the client, maybe we should do
```suggestion
type CloudProviderAccessRoleDeauthorizer interface {
	DeauthorizeCloudProviderAccessRoles(*atlas.CloudProviderDeauthorizationRequest) error
}
```